student.c

```diff
@@ -1,20 +1,42 @@
-#include <assert.h>
 #include "student.h"
 
+#include <assert.h>
+
+static ll_comparator_t compare_gpa(struct ll_node *first,
+                                   struct ll_node *second) {
+    struct student *first_student = LL_ENTRY(first, struct student, node);
+    struct student *second_student = LL_ENTRY(second, struct student, node);
+    if (first_student->gpa < second_student->gpa) {
+        return -1;
+    } else if (first_student->gpa > second_student->gpa) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+static ll_predicate_t is_honour_roll(struct ll_node *cur_node) {
```

Comment:
Same remark as for `compare_gpa`: the return type should be `bool`, not `ll_predicate_t`.